File: #1-25 (C, Java, Python)/Java/Euler3.java
//Euler3 ~ Largest prime factor 
/* Euler requirements limit runtime to 60s. Limiting the runtime
   to 60s provides the correct answer in this case, but for other
   values this may not be true. In those cases, a precompiled list of
   primes can be created, circumventing the prime check for all values
   less than N/2. Alternatively, a more robust primality test can
   be implemented.*/
import java.util.concurrent.TimeUnit;

public class Euler3
{
    public static void main(String[] args) throws InterruptedException
	{
		long startTime    = System.nanoTime();
		long endTime      = 0;
		long timeElapsed  = 0;
		long runTime      = 0;
		long largestPrime = 1;
		long val          = 600851475143L;
		
		for(long i = 1; i < val/2; i++)
		{	
			if(runTime >= 60)
			{
				System.out.println("Runtime has exceeded requirement.\n");
				System.out.println("Possible solution: " + largestPrime + "\n");
				System.out.println("Program completed in " + timeElapsed/1000000000.0 + " seconds.");
				return;
			}
			
			if((val%i)==0)
			{
				if(primeCheck(i))
				{
					largestPrime = i;
				}
			}
			endTime = System.nanoTime();
			timeElapsed = endTime - startTime;
			runTime = timeElapsed/1000000000;
		}
		
		/* endTime = System.nanoTime();
		timeElapsed = endTime - startTime;
		System.out.println("Solution: " + largestPrime + "\n");
		System.out.println("Program completed in " + timeElapsed/1000000000.0 + " seconds."); */
    }
	
	//Very unoptimized implementation of primality test
	static boolean primeCheck(long n)
	{	
		for(long j = 2; j <= (n-1); j++)
		{
			if((n%j)==0)
			{
				return false;
			}
		}
		return true;
	}
}

File: #1-25 (C, Java, Python)/Python/Euler3.py
#Euler3 ~ Largest prime factor
import timeit
start = timeit.default_timer()

cpu_time     = 0
largestPrime = 1
val          = 600851475143

#Very unoptimized implementation of primality test
def is_prime_number(x):
    if x >= 2:
        for y in range(2,x):
            if not ( x % y ):
                return False
    else:
        return False
    return True

r = int(val/2)
for i in range(1,r):
    stop = timeit.default_timer()
    cpu_time = stop - start
    if(cpu_time > 60):
        print("Runtime has exceeded requirement.\n")
        print("Possible solution:",largestPrime,"\n")
        print("Program completed in",cpu_time,"seconds.")
        exit()
        
    if((val%i) == 0):
        if is_prime_number(i):
            largestPrime = i
            
            
#stop = timeit.default_timer()
# print("Solution: ","\n")
# print("Program completed in","{0:.3f}".format(stop - start), "seconds.")